vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location.live
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.updateLayoutParams
+import im.vector.app.R
+import im.vector.app.databinding.ViewLocationLiveEndedBannerBinding
+
+class LocationLiveEndedBannerView @JvmOverloads constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding = ViewLocationLiveEndedBannerBinding.inflate(
+            LayoutInflater.from(context),
+            this
+    )
+
+    init {
+        context.theme.obtainStyledAttributes(
+                attrs,
+                R.styleable.LocationLiveEndedBannerView,
+                0,
+                0
+        ).run {
+            try {
+                setBackgroundAlpha(this)
+                setIconMarginStart(this)
+            } finally {
+                recycle()
+            }
+        }
```

Comment:
I think you can simplify like this
```suggestion
        context.obtainStyledAttributes(
                attrs,
                R.styleable.LocationLiveEndedBannerView,
                0,
                0
        ).use {
                setBackgroundAlpha(it)
                setIconMarginStart(it)
        }
```

Reply:
Yes! Thanks for the suggestion, I didn't see there were the extension `use` for `TypedArray`.

Reply:
Updated, see: [d62cf70](https://github.com/vector-im/element-android/pull/6636/commits/d62cf70b8c01496cc07acca12b57f41b2d7b9e1a)

Reply:
Not an issue but you missed the shortcut `context.theme.obtainStyledAttributes` -> `context.obtainStyledAttributes`

Reply:
Yes indeed, thanks. Updated here: [0750e80](https://github.com/vector-im/element-android/pull/6636/commits/0750e80f04918b9a0e4649193a69be805cd032c9)